src/content-tags-drawer/ContentTagsCollapsible.jsx

```diff
@@ -131,66 +195,58 @@ const ContentTagsCollapsible = ({ contentId, taxonomyAndTagsData }) => {
     }
   }, []);
 
-  const modalPopupOnCloseHandler = React.useCallback((event) => {
-    close(event);
-    // Clear search term
-    setSearchTerm('');
-  }, []);
+  // onChange handler for react-select component, currently only called when
+  // staged tags in the react-select input are removed or fully cleared.
+  // The remaining staged tags are passed in as the parameter, so we set the state
+  // to the passed in tags
+  const handleStagedTagsMenuChange = React.useCallback((stagedTags) => {
+    const prevStagedContentTags = cloneDeep(stagedContentTags);
```

Comment:
I don't think you need this `cloneDeep`, as the value of `stagedContentTags` is not going to change during the lifetime of this function. Especially if you can move `setStagedTags()` to the end.

Reply:
Good point, I've removed the cloneDeep and moved the setStagedTags to the end.